fairscale/optim/oss.py

```diff
@@ -624,10 +609,24 @@ def _setup_bucket_strategy(self) -> None:
         network requests have been issued.
         """
 
-        # Determine the max work handles in flight:
-        # - count all the buckets on the fly
-        self._max_work_handles = 0
+        # (re) allocate the buckets
+        #  - Get the correct size for the buckets, cannot be bigger than the model
+        model_size = sum([p.numel() for p in self.param_to_rank.keys()])
```

Comment:
re-allocating the buckets from scratch (original size) anytime the bucketing strategy is updated